Automattic/jetpack
# General: Add ability to preview a post while logged out

Currently, if a user is in Calypso and previews a post (while the user is logged out of the Jetpack site), then the post preview 404s. 😞 

This PR makes use of the frame nonce that is passed from WPCOM to display the preview.

To test:
- On WPCOM, checkout D2472
- Sandbox `public-api.wordpress.com`
- In Jetpack, set API base to your sandbox
- On `$site` (Jetpack site), checkout `update/calypso-post-previews` branch
- Log out of `$site`
- Go to `wordpress.com/post/$site`
- Create a draft
- Click preview button
- Verify that you can see the preview

Note: You may need to wait for the nonce to update on Calypso. This should only be a couple of minutes. Alternatively, just clear state and localStorage.

Note: This PR does not handle viewing the preview within an iFrame. That is going to be a much more involved process with security concerns.

cc @mtias who brought this up to Poseidon
cc @lezama for code review
